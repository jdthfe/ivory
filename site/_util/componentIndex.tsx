export default [
    { name: 'CheckUA', type: 'Function', 'zh-CN': '校验UA' },
    { name: 'DeepObjectMerge', type: 'Function', 'zh-CN': '数据扁平化' },
    { name: 'FormatePrice', type: 'Function', 'zh-CN': '格式化价钱' },
    { name: 'GetQuery', type: 'Function', 'zh-CN': '解析URL' },
    { name: 'Throttle', type: 'Function', 'zh-CN': '待输入_中文名' },
];
