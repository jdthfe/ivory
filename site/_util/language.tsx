// en-US must be the first item
export interface LanguageItem {
    code: Language;
    content: string;
}
export const languages: LanguageItem[] = [
    { code: 'en-US', content: 'English' },
    { code: 'zh-CN', content: '中文' },
];

/** 判断语言 */
export function getLanguage(): Language {
    const code = String(localStorage.getItem('language')) as Language;
    if (languages.map((item) => item.code).includes(code)) {
        return code;
    } else {
        return 'en-US';
    }
}
