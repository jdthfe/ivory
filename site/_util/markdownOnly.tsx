export default [
    { name: 'Introduce', type: 'markdownOnly', 'zh-CN': '介绍' },
    { name: 'Customize Theme', type: 'markdownOnly', 'zh-CN': '主题' },
    { name: '_util', type: 'markdownOnly', 'zh-CN': '工具' },
];
