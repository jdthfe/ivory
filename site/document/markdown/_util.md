Common variables in components

```tsx
export const prefix = 'eui';
export const Variable = {
    defaultDuration: 2000,
    transitionTime: 160,
    transitionFade: `${prefix}-fade`,
    transitionZoom: `${prefix}-zoom`,
};

export const removeNode = (el: HTMLDivElement) => {
    if (el.remove) {
        el.remove();
    } else if (el.parentNode) {
        el.parentNode.removeChild(el);
    }
};

export default {
    prefix,
    Variable,
    removeNode,
};
```

Usage

```tsx
import { _util } from '@jdthfe/eui';
// import _util from 'eui/compiled/lib/_util/'; // If you use babel-plugin-import

const { Variable } = _util;
console.log(Variable.defaultDuration); // 2000
```
