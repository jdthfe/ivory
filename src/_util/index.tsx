export const prefix = 'eui';
export const Variable = {
    defaultDuration: 2000,
    transitionTime: 160,
    transitionFade: `${prefix}-fade`,
    transitionZoom: `${prefix}-zoom`,
};

export const removeNode = (el: HTMLDivElement) => {
    if (el.remove) {
        el.remove();
    } else if (el.parentNode) {
        el.parentNode.removeChild(el);
    }
};

export default {
    prefix,
    Variable,
    removeNode,
};
