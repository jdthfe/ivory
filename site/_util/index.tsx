import componentIndexNoType from './componentIndex';
import markdownOnlyNoType from './markdownOnly';
import { languages } from './language';

export interface Docs {
    readme: string;
    'readme.zh-CN'?: string;
    demoSource?: string;
    [key: string]: string | undefined;
}

export const componentIndex: ComponentIndex[] = componentIndexNoType;
export const markdownOnly: ComponentIndex[] = markdownOnlyNoType;

export const sortComponentIndex = (componentIndex: ComponentIndex[]) => {
    const newStructure: { [propName: string]: ComponentIndex[] } = {};
    componentIndex.map((item) => {
        if (newStructure[item.type]) {
            newStructure[item.type].push(item);
        } else {
            newStructure[item.type] = [item];
        }
    });
    return newStructure;
};
export const getInstance = (item: ComponentIndex) => {
    let instance: React.FC;
    try {
        if (item.type === 'markdownOnly') {
            throw 'markdownOnly have not Instance';
        } else {
            instance = require(`../../src/${item.name}/demo`).default;
        }
    } catch (e) {
        throw e;
    }
    return instance;
};
export const getDocs = (item: ComponentIndex) => {
    const docs: Docs = { readme: '' };
    try {
        if (item.type === 'markdownOnly') {
            languages.map((lang) => {
                const readmeLang = `readme${
                        lang.code === 'en-US' ? '' : '.' + lang.code
                    }`,
                    mdLang = `${item.name}${
                        lang.code === 'en-US' ? '' : '.' + lang.code
                    }`;
                docs[readmeLang] =
                    require(`!!raw-loader!../document/markdown/${mdLang}.md`).default;
            });
        } else {
            docs.demoSource =
                require(`!!raw-loader!../../src/${item.name}/demo`).default;
            languages.map((lang) => {
                const readmeLang = `readme${
                    lang.code === 'en-US' ? '' : '.' + lang.code
                }`;
                docs[readmeLang] =
                    require(`!!raw-loader!../../src/${item.name}/demo/${readmeLang}.md`).default;
            });
        }
    } catch (e) {
        throw e;
    }
    return docs;
};
