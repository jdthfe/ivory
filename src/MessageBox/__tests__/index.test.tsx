import { MessageBox } from '@src/index';
import React from 'react';
import { render, waitForElement, waitForElementToBeRemoved, fireEvent } from 'react-testing-library';
import { prefix } from '../../_util/';

function clearSvg(baseElement: HTMLElement) {
    //  排除 svg 文件干扰
    const svg = baseElement.querySelector('svg');
    if (svg) {
        svg.remove();
    }
}

describe('MessageBox', () => {
    const MessageBoxText = 'MessageBoxText';
    const testDiv = <div className="testDiv" />;
    const { baseElement, getByText } = render(testDiv);

    it('snapshots model', async () => {
        const close = MessageBox.model({ children: MessageBoxText });
        await waitForElement(() => getByText(MessageBoxText));
        expect(baseElement).toMatchSnapshot('model show');
        close();
        await waitForElementToBeRemoved(() => getByText(MessageBoxText));
        expect(baseElement).toMatchSnapshot('model close');
    });

    it('default model onClickCloseIcon', async () => {
        const onClick = jest.fn();

        MessageBox.model({ children: MessageBoxText, onClickCloseIcon: onClick });

        await waitForElement(() => getByText(MessageBoxText));
        fireEvent.click(baseElement.querySelector(`.${prefix}-messagebox-close`)!);
        await waitForElementToBeRemoved(() => getByText(MessageBoxText));

        expect(onClick).toBeCalled();
    });

    it('default model buttons', async () => {
        const confirmChildren = 'confirmChildren';
        const coverClassName = 'coverClassName';
        const onClick = jest.fn();

        const close = MessageBox.model({
            onClickCover: () => false,
            children: MessageBoxText,
            coverProps: { className: coverClassName },
            buttons: [
                {
                    onClick: () => {
                        onClick();
                        return false;
                    },
                    children: confirmChildren,
                },
            ],
        });

        await waitForElement(() => getByText(MessageBoxText));
        fireEvent.click(getByText(confirmChildren));
        await waitForElement(() => getByText(MessageBoxText));
        fireEvent.click(baseElement.querySelector(`.${coverClassName}`)!);
        await waitForElement(() => getByText(MessageBoxText));
        close();
        await waitForElementToBeRemoved(() => getByText(MessageBoxText));
        expect(onClick).toBeCalled();
    });

    it('default model onClickCover', async () => {
        const onClick = jest.fn();
        const coverClassName = 'coverClassName';

        MessageBox.model({ children: MessageBoxText, coverProps: { onClick, className: coverClassName } });

        await waitForElement(() => getByText(MessageBoxText));
        fireEvent.click(baseElement.querySelector(`.${coverClassName}`)!);
        await waitForElementToBeRemoved(() => getByText(MessageBoxText));

        expect(onClick).toBeCalled();
    });

    it('snapshots alert', async () => {
        const confirmCallback = jest.fn();
        const onClick = jest.fn();
        const confirmChildren = 'confirmChildren';

        MessageBox.alert({
            multiLineButtons: true,
            title: 'title',
            confirmButton: { onClick },
            children: MessageBoxText,
            confirmChildren,
            confirmCallback,
        });
        await waitForElement(() => getByText(confirmChildren));
        clearSvg(baseElement);
        expect(baseElement).toMatchSnapshot('alert show');
        fireEvent.click(getByText(confirmChildren));
        await waitForElementToBeRemoved(() => getByText(MessageBoxText));
        expect(baseElement).toMatchSnapshot('alert close');
        expect(confirmCallback).toBeCalled();
        expect(onClick).toBeCalled();
    });

    it('default alert', async () => {
        const confirmChildren = 'confirm';
        MessageBox.alert();
        await waitForElement(() => getByText(confirmChildren));
        fireEvent.click(getByText(confirmChildren));
        await waitForElementToBeRemoved(() => getByText(confirmChildren));
        expect(true).toEqual(true);
    });
    it('snapshots confirm', async () => {
        const confirmCallback = jest.fn();
        const onClick = jest.fn();
        const confirmChildren = 'confirmChildren';

        MessageBox.confirm({
            multiLineButtons: true,
            title: 'title',
            confirmButton: { onClick },
            children: MessageBoxText,
            confirmChildren,
            confirmCallback,
        });
        await waitForElement(() => getByText(confirmChildren));
        expect(baseElement).toMatchSnapshot('confirm show');
        fireEvent.click(getByText(confirmChildren));
        await waitForElementToBeRemoved(() => getByText(MessageBoxText));
        expect(baseElement).toMatchSnapshot('confirm close');
        expect(confirmCallback).toBeCalled();
        expect(onClick).toBeCalled();
    });

    it('default confirm', async () => {
        const cancelChildren = 'cancel';
        MessageBox.confirm();
        await waitForElement(() => getByText(cancelChildren));
        fireEvent.click(getByText(cancelChildren));
        await waitForElementToBeRemoved(() => getByText(cancelChildren));
        expect(true).toEqual(true);
    });
    it('cancel confirm & noCover', async () => {
        const onClick = jest.fn();
        const cancelChildren = 'cancel';
        const coverClassName = 'coverClassName';
        MessageBox.confirm({
            onClickCover: false,
            cancelButton: { onClick },
            coverProps: { className: coverClassName },
        });
        await waitForElement(() => getByText(cancelChildren));
        expect(baseElement.querySelector(`.${coverClassName}`) === null).toEqual(true);
        fireEvent.click(getByText(cancelChildren));
        await waitForElementToBeRemoved(() => getByText(cancelChildren));
        expect(onClick).toBeCalled();
    });
});
