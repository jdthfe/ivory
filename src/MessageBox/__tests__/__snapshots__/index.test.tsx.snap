// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox snapshots alert: alert close 1`] = `
<body>
  
  
  

  <div>
    <div
      class="testDiv"
    />
  </div>
</body>
`;

exports[`MessageBox snapshots alert: alert show 1`] = `
<body>
  
  
  

  <div>
    <div
      class="testDiv"
    />
  </div>
  <div>
    <div
      class="eui-cover eui-fade-entry-active eui-fade"
    />
    <div
      class="eui-messagebox undefined eui-messagebox-buttons-multiline-single eui-fade-entry-active eui-fade"
    >
      <p
        class="eui-messagebox-title"
      >
        title
      </p>
      <div
        class="eui-messagebox-content"
      >
        MessageBoxText
      </div>
      <div
        class="eui-messagebox-buttons eui-messagebox-buttons-multiline"
      >
        <button
          class="eui-button eui-button-l eui-button-primary"
          role="button"
        >
          confirmChildren
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`MessageBox snapshots confirm: confirm close 1`] = `
<body>
  
  
  

  <div>
    <div
      class="testDiv"
    />
  </div>
</body>
`;

exports[`MessageBox snapshots confirm: confirm show 1`] = `
<body>
  
  
  

  <div>
    <div
      class="testDiv"
    />
  </div>
  <div>
    <div
      class="eui-cover eui-fade-entry-active eui-fade"
    />
    <div
      class="eui-messagebox eui-fade-entry-active eui-fade"
    >
      <p
        class="eui-messagebox-title"
      >
        title
      </p>
      <div
        class="eui-messagebox-content"
      >
        MessageBoxText
      </div>
      <div
        class="eui-messagebox-buttons eui-messagebox-buttons-multiline"
      >
        <button
          class="eui-button eui-button-l eui-button-primary"
          role="button"
        >
          confirmChildren
        </button>
        <button
          class="eui-button eui-messagebox-buttons-multiline-last eui-button-l eui-button-customize eui-button-ghost"
          role="button"
        >
          cancel
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`MessageBox snapshots model: model close 1`] = `
<body>
  <div>
    <div
      class="testDiv"
    />
  </div>
</body>
`;

exports[`MessageBox snapshots model: model show 1`] = `
<body>
  <div>
    <div
      class="testDiv"
    />
  </div>
  <div>
    <div
      class="eui-cover eui-fade-entry-active eui-fade"
    />
    <div
      class="eui-messagebox eui-fade-entry-active eui-fade"
    >
      <div
        class="eui-messagebox-content"
      >
        MessageBoxText
      </div>
      <div
        class="eui-messagebox-line"
      />
    </div>
  </div>
</body>
`;
